src/buffer.c

```diff
@@ -83,7 +85,26 @@ size_t qd_buffer_size(qd_buffer_t *buf)
 void qd_buffer_insert(qd_buffer_t *buf, size_t len)
 {
     buf->size += len;
-    assert(buf->size <= buffer_size);
+    assert(buf->size <= BUFFER_SIZE);
+}
+
+void qd_buffer_add_fanout(qd_buffer_t *buf)
+{
+    buf->fanout++;
```

Comment:
Currently, the only function calling qd_buffer_add_fanout() is qdr_forward_new_delivery_CT(in forwarder.c) which is only called by the core thread. So, there should be no contention. But I am happy to change the fanout to atomic so that anybody else calling it outside of the core thread will not be affected.